backend/routes.js

```diff
@@ -39,25 +39,68 @@ route.POST('/event', {
+      if (deserializedHEAD.contractID === deserializedHEAD.hash) {
+        // Store attribution information
+        if (credentials?.billableContractID) {
+          // Store the owner for the current resource
+          await sbp('chelonia/db/set', `_private_owner_${deserializedHEAD.contractID}`, credentials.billableContractID)
+          const resourcesKey = `_private_resources_${credentials.billableContractID}`
+          // Store the resource in the resource index key
+          // This is done in a queue to handle several simultaneous requests
+          // reading and writing to the same key
+          await sbp('okTurtles.eventQueue/queueEvent', resourcesKey, async () => {
+            const existingResources = await sbp('chelonia/db/get', resourcesKey)
+            await sbp('chelonia/db/set', resourcesKey, (existingResources ? existingResources + '\x00' : '') + deserializedHEAD.contractID)
+          })
+        // A billable entity has been created
+        } else {
+          // Use a queue to ensure atomic updates
+          await sbp('okTurtles.eventQueue/queueEvent', '_private_billable_entities', async () => {
+            const existingBillableEntities = await sbp('chelonia/db/get', '_private_billable_entities')
+            await sbp('chelonia/db/set', '_private_billable_entities', (existingBillableEntities ? existingBillableEntities + '\x00' : '') + deserializedHEAD.contractID)
+          })
+        }
+        // If this is the first message in a contract and the
+        // `shelter-namespace-registration` header is present, proceed with also
+        // registering a name for the new contract
+        const name = request.headers['shelter-namespace-registration']
+        if (name && !name.startsWith('_private')) {
         // Name registation is enabled only for identity contracts
-        const cheloniaState = sbp('chelonia/private/state')
-        if (cheloniaState.contracts[deserializedHEAD.contractID]?.type === 'gi.contracts/identity') {
-          const r = await sbp('backend/db/registerName', name, deserializedHEAD.contractID)
-          if (Boom.isBoom(r)) {
-            return r
+          const cheloniaState = sbp('chelonia/rootState')
+          if (cheloniaState.contracts[deserializedHEAD.contractID]?.type === 'gi.contracts/identity') {
+            const r = await sbp('backend/db/registerName', name, deserializedHEAD.contractID)
+            if (Boom.isBoom(r)) {
+              return r
+            }
           }
         }
       }
+      // Store size information
+      const sizeKey = `_private_size_${deserializedHEAD.contractID}`
+      // Use a queue to ensure atomic updates
+      await sbp('okTurtles.eventQueue/queueEvent', sizeKey, async () => {
+        // Size is stored as a hex value
+        const existingSize = parseInt(await sbp('chelonia/db/get', sizeKey, 16)) || 0
+        await sbp('chelonia/db/set', sizeKey, (existingSize + Buffer.byteLength(request.payload)).toString(16))
```

Comment:
Why is the size stored as a hex value? Seems simpler and more natural (for humans) for it to be stored as a regular decimal?

Reply:
Sure. This will likely not be seen by humans directly and hex is slightly more efficient to parse (that was the reason)